include/util/shared_memory_vector_wrapper.hpp

```diff
@@ -80,6 +81,18 @@ template <typename DataT> class SharedMemoryWrapper
         m_size = size;
     }
 
+    // does nothing but ensure we don't call invalud vector-like
+    // operations on this object
+    void resize(std::size_t size) const
+    {
+        if (m_size != size)
+        {
+            throw util::exception("Invalid resize " + std::to_string(size) +
+                                  " on immutable vector view of size " + std::to_string(m_size) +
+                                  ".");
+        }
+    }
```

Comment:
For comparability with the `std::vector` interface code. With that we can use the same code for de-serializing a vector and a vector view (resize is NOP on a vector that is already of correct size and a NOP here too).